.changes/unreleased/operator-Changed-20250605-122221.yaml

```diff
@@ -0,0 +1,4 @@
+project: operator
+kind: Changed
+body: The v2 operator will now unconditionally re-queue for reconciliation periodically. This permits it to pick up configuration changes in external secrets. The same beahviour is already present in the v1 operator.
```

Comment:
nit: Changelogs should be primarily targeted towards selfhosted users. Even though the external secret behavior is not technically user facing just yet 😓 

```suggestion
body: The `Redpanda` controller will now unconditionally re-queue for reconciliation periodically. This permits it to pick up configuration changes in external secrets.
```